src/main/java/lesson2/homework/Main.java

```diff
@@ -0,0 +1,95 @@
+package lesson2.homework;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.Random;
+import java.util.TreeSet;
+
+public class Main {
+
+  public static void main(String[] args) {
+    Random random = new Random();
+    int[] testArray = new int[100_000];
+    for (int i = 0; i < testArray.length; i++) {
+      testArray[i] = random.nextInt();
+    }
+    System.out.println();
+
+
+   // Array<Integer> data = new ImplArray<>();
```

Comment:
Не смог разобраться с проблемой "Can not infer arguments". Вроде дляю все правильно, не понимаю. Дженерики это сложно(

Reply:
Скорее всего вы пытались передать в конструктор массив примитивов, а внутри ожидается ссылочный тип Integer. Правда, у вас конструктор приватный, поэтому не совсем понимаю, на каком кейсе компилятор выдает вам ошибку

Reply:
Просто мне тоже не понятно. Строго говоря все что написано  
 ImplArray, ActionsOnArray<E>  это копия вашего кода с небольшими изменениями.
ошибка появяется когда раскомментрую строку
Array<Integer> data = new ImplArray<>();

Error:(20, 37) java: incompatible types: cannot infer type arguments for lesson2.homework.ImplArray<>
    reason: no instance(s) of type variable(s) E exist so that lesson2.homework.ImplArray<E> conforms to lesson2.homework.Array<java.lang.Integer>
Гугл невнятно объясняет(